libpod/container_api.go

```diff
@@ -678,3 +679,101 @@ func (c *Container) Sync() error {
 
 	return nil
 }
+
+// Checkpoint checkpoints a container
+func (c *Container) Checkpoint() error {
+	if !c.locked {
+		c.lock.Lock()
+		defer c.lock.Unlock()
+
+		if err := c.syncContainer(); err != nil {
+			return err
+		}
+	}
+
+	if c.state.State != ContainerStateRunning {
+		return errors.Wrapf(ErrCtrStateInvalid, "%q is not running, cannot checkpoint", c.state.State)
+	}
+	if err := c.runtime.ociRuntime.checkpointContainer(c); err != nil {
+		return err
+	}
+
+	logrus.Debugf("Checkpointed container %s", c.ID())
+
+	c.state.State = ContainerStateStopped
+
+	// This needs to be deleted so that the restore works correctly
+	logrus.Debugf("Cleaning up %s", filepath.Join(c.bundlePath(), "attach"))
+	os.Remove(filepath.Join(c.bundlePath(), "attach"))
+	logrus.Debugf("Cleaning up %s", filepath.Join(c.bundlePath(), "ctl"))
+	os.Remove(filepath.Join(c.bundlePath(), "ctl"))
+
+	if err := c.cleanupStorage(); err != nil {
+		return err
+	}
+
+	return c.save()
+}
+
+// Restore restores a container
+func (c *Container) Restore() error {
+	if !c.locked {
+		c.lock.Lock()
+		defer c.lock.Unlock()
+
+		// This does not work (yet)
+		//if err := c.syncContainer(); err != nil {
+		//	return err
+		//}
+	}
+
+	if c.state.State == ContainerStateRunning {
```

Comment:
Probably worth including Paused in there too, I doubt we want to restore over a paused container